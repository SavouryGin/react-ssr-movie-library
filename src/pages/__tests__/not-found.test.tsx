import NotFoundPage from 'pages/not-found';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

describe('NotFoundPage page:', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <NotFoundPage />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains the error title and the link', () => {
    render(
      <BrowserRouter>
        <NotFoundPage />
      </BrowserRouter>,
    );
    const header = screen.getByRole('heading');
    const link = screen.getByRole('link');
    expect(header).toBeInTheDocument();
    expect(header).toHaveTextContent('404 Error');
    expect(link).toBeInTheDocument();
    expect(link).toHaveTextContent('Go Home');
  });
});
