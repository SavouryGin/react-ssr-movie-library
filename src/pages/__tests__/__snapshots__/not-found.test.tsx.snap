// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage page: matches the snapshot 1`] = `
<DocumentFragment>
  <article>
    <h2>
      404 Error
    </h2>
    <p>
      Sorry, your page was not found. 
      <a
        href="/"
      >
        Go Home
      </a>
    </p>
  </article>
</DocumentFragment>
`;
